### DISCUSSION TITLE
configuring with oidc issues

### DISCUSSION THREAD

**Discussion by Pshemas** [Ylianst/MeshCentral#6112] [Q&A] [answered]
I'm trying to set up MeshCentral with Authentik oidc provider. When I try to log in with oidc I get:

```
ERR: TypeError: Cannot read properties of undefined (reading 'claim')
    at OpenIDConnectStrategy.oidcCallback [as _verify] (/home/meshcentral/MC/node_modules/meshcentral/webserver.js:7471:63)
    at /home/meshcentral/MC/node_modules/openid-client/lib/passport_strategy.js:189:10
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
```

shortly after:

```
WEBREQUEST: (some.ip) /auth-oidc-callback?code=...
```

Atm the config for oidc is simple:

```
        "oidc": {
          "issuer": "https://aaaaap/application/o/meshcentral/",
          "clientid": "CID from Authentik",
          "clientsecret": "CIS from Authentik",
          "newAccounts": "true"
        }
```

I've tried more complex config first, but as that didn't work either I've moved to basic one. 
While running MC with `--debug` I see 

```
AUTHLOG: OIDC: Discovering Issuer Endpoints: https://myendpoint
VERBOSE: OIDC: Saved Configuration: {"client": ....
AUTHLOG: OIDC: Setup Complete
AUTHLOG: Setting up authentication strategies login and callback URLs for root domain.
AUTHLOG: OIDC: Authorization URL: /auth-oidc
AUTHLOG: OIDC: Callback URL: /auth-oidc-callback
HTTPS: Server listening on 0.0.0.0 port 443.

```

Any ideas what am I doing wrong?

**Comment by si458** [comment, marked as answer]
the is problems with OIDC in version 1.1.21+ due to a bad merge from a community member,
BUT ive spent the week fixing the ODIC issues and these have been fixed in the master branch now!
just waiting to do a new release
but in the mean time 
if you can use the `master` docker image
OR
stop meshcentral, **backup everything including node_modules**, `npm install Ylianst/MeshCentral`, restart meshcentral

https://github.com/Ylianst/MeshCentral/issues/6104

EDIT:
the is also new docs here for the oidc which the community member did which is helpful
https://ylianst.github.io/MeshCentral/meshcentral/openidConnectStrategy/

**Comment by Pshemas** [comment]
thank you @si458 , that helped. Localization seems to be gone, but that's not a big deal.
BTW - yeah, I know this docs, that's how I've set it up initially. The docs are one of the key factors that make me like MC so much. They're awesome and kudos to Ylian for going extra mile to even provide videos.

**Reply by si458** [reply]
glad the master branch/docker image helped!
yes to fix the localization, you need to do the following
(im guessing u used the `npm install` method as the docker image should be ok for translations)
1. `cd node_modules/meshcentral/translate`
2. `node translate.js` <- WAIT AS THIS WILL INSTALL MISSING PACKAGES AND IT SHOULD ERROR OUT WHICH IS OK
3. `node translate.js` <- run again to check it outputs the help information now
4. `node translate.js extractall` <- this extracts all english string from the view files ready to translate
5. `node translate.js translateall` <- this will now translate everything (sit back n have a brew takes about 10mins)
6. restart your meshcentral with `resetserver` in the console OR `systemctl` OR however you stop/start ur MC

**Reply by Pshemas** [reply]
yeah, I've used `npm` method. Glad to hear container should be even more hassle-free. I plan to move it to another machine, where most of the things are run with Podman.
